damnit/ctxsupport/ctxrunner.py

```diff
@@ -45,6 +47,72 @@ class DataType(Enum):
     Timestamp = "timestamp"
 
 
+class MyMetadataClient:
+    def __init__(self, proposal, timeout=10, init_server="https://exfldadev01.desy.de/zwop"):
+        self.proposal = proposal
+        self.timeout = timeout
+        self._cache = {}
```

Comment:
With how the cache is used (keys as a tuple of arguments basically as far as I could tell?) it seems pretty similar to what using the functools `cache` decorator would do right?

e.g. 

```python
    @cache
    def _run_info(self, run):
        response = requests.get(f"{self.server}/api/mymdc/proposals/by_number/{self.proposal}/runs/{run}",
                                    headers=self._headers, timeout=self.timeout)
        ....
```

Reply:
That's true, but when we make this public the caching should probably be disabled by default (at least until mymdc implements caching headers), so I'll stick with the dictionary for now so we can add an option to disable caching later.